### PR TITLE
assert(): Make message argument optional

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

**Comment by xclaesse** [comment]
This depends on https://github.com/mesonbuild/meson/pull/6281.

File: run_unittests.py [hunk 1]
```diff
@@ -4386,6 +4386,14 @@ def test_version_checked_before_parsing_options(self):
...
         self.assertMesonRaises("", match, meson_version='>=2000', options=options)
 
+    def test_assert_default_message(self):
+        self.assertMesonRaises('''
+            k1 = 'a'
+            assert({
+              k1: 1,
+              'b': 2
+            }['a'] == 2)''',
+            "Assert failed: {k1 : 1, 'b' : 2}\['a'\] == 2")
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ continuation line under-indented for visual indent ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6285#issue-4893477))

<details>

| Rule |
| - |
| E128 |

</details>
<sub>posted by Sider</sub>

Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ invalid escape sequence '\[' ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6285#issue-4893478))

<details>

| Rule |
| - |
| W605 |

</details>
<sub>posted by Sider</sub>

File: run_unittests.py [hunk 2]
```diff
@@ -4386,6 +4386,15 @@ def test_version_checked_before_parsing_options(self):
...
 
+    def test_assert_default_message(self):
+        self.assertMesonRaises('''
+                               k1 = 'a'
+                               assert({
+                                 k1: 1,
+                                 'b': 2
+                               }['a'] == 2)
+                               ''',
+                               "Assert failed: {k1 : 1, 'b' : 2}\['a'\] == 2")
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ invalid escape sequence '\[' ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6285#issue-4893479))

<details>

| Rule |
| - |
| W605 |

</details>
<sub>posted by Sider</sub>

**Comment by jpakkane** [comment]
Trying to rerun the CI from the web page does not work. Please rebase.

**Comment by xclaesse** [comment]
I retried the CI multiple times already, it consistently fail the same way. But it makes no sense to me, why only that specific setup fails and all others works? That sounds like a bug in python itself, is that possible? Maybe update python in the docker?

@jpakkane I'm totally clueless on this failure, any idea?

**Comment by jpakkane** [comment]
Weird. The only thing I can think of is that the string `assert` is confusing it. Maybe try renaming it to `test_default_asrt_message`?

**Comment by jpakkane** [comment]
Things get even weirder now. But at least it's not related to this:

```
Locating Python:
C:\Python27\python.exe
Python 2.7.3
C:\Python27\python.exe: No module named pip
```

Python 2? How is it possible?

**Comment by xclaesse** [comment]
@jpakkane I've seen that on other PRs too.

Renaming the function without 'assert' didn't fix the issue :( At this point the only way I can think about is skipping the test on Windows.

**Comment by mensinda** [comment]
@xclaesse I also ran into the INTERNALERROR issue with xdist on my machine locally a while back. I was able to debug what was actually happening by reverting back to unittest (just add `raise ImportError()` [here](https://github.com/mesonbuild/meson/blob/master/run_unittests.py#L7020)).

@jpakkane In general, the msvc2015 builds are all failing because only python2 is in the PATH. The real error is that choco is unable to install python3:

```
Chocolatey v0.10.15
Installing the following packages:
python3
By installing you accept licenses for the packages.
python3 not installed. An error occurred during installation:
 Unable to resolve dependency 'chocolatey-core.extension (≥ 1.3.3)'.
python3 package files install completed. Performing other installation steps.
The install of python3 was NOT successful.
python3 not installed. An error occurred during installation:
 Unable to resolve dependency 'chocolatey-core.extension (≥ 1.3.3)'.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - python3 (exited 1) - python3 not installed. An error occurred during installation:
 Unable to resolve dependency 'chocolatey-core.extension (≥ 1.3.3)'.
```

So all non `[skip ci]` builds will fail in msvc2015 until this is fixed (yay for Azure reliabilety).

**Comment by xclaesse** [comment]
@mensinda Thanks for the hint, it's happening in https://github.com/mesonbuild/meson/pull/6203 too now. I guess pytest and/or pytest-xdist were not installed on the runner, and probably now they are installed which would explain why it suddenly start failing.

**Comment by xclaesse** [comment]
Sounds like that bug: https://github.com/pytest-dev/pytest-xdist/issues/204

**Comment by xclaesse** [comment]
@mensinda @jpakkane I added a commit that disable pytest on python <= 3.5 because it seems to work with >=3.6,  at least that's what I have locally to test.

**Comment by mensinda** [comment]
Let's hope it works. And #6319 will hopefully resolve the Chocolatey issues in vs2015 (that may or may not be present anymore and/or appear again).

**Pull request merged**